web-app/src/services/holidayService.js

```diff
@@ -24,8 +24,16 @@ export const approveHoliday = eventId => {
   return axios.put('/holidays/approveHoliday', { eventId });
 };
 
-export const rejectHoliday = eventId => {
-  return axios.put('/holidays/rejectHoliday', { eventId });
+export const rejectHoliday = holidayId => {
+  return axios.put('/holidays/cancelHoliday', { holidayId });
+};
+
+export const rejectHolidayWithMessage = ( eventId, message ) => {
+  return axios.put('/holidays/rejectHoliday', { eventId, message });
+};
```

Comment:
Just have this as rejectHoliday but with an optional message parameter & split the behaviour by checking to see if the message exists.